content/vi/docs/concepts/architecture/cloud-controller.md

```diff
@@ -0,0 +1,241 @@
+{{< /note >}}
+
+Kết hoạch ban đầu của dự án là hỗ trợ Volume sử dụng bộ Cloud Controller Manager để áp dụng những Volume linh hoạt nhằm dễ dàng tích hợp bổ sung thêm. Tuy nhiên, một giải pháp khác cũng đang được lên kế hoạch để thay thế Volume linh hoạt được biết là CSI.
+
+Sau khi xem xét về khía cạnh này, chúng tôi quyết định có một khoảng dừng đo đạc trung gian trước khi CSI trở nên sẵn sàng cho việc sử dụng.
+
+## Chức năng của bộ Cloud Controller Manager
+
+Bộ CCM thừa hưởng những tính năng của nó từ các thành phần trong Kubernetes phụ thuộc vào các Cloud Provider. Phần kế tiếp sẽ giới thiệu những thành phần này.
+
+### 1. Kubernetes Conntroller Manager
+
+Phần lớn các tính năng của bộ CCM bắt nguồn từ Kubernetes controller manager. Như đã đề cập ở phần trước, bộ CCM bao gồm:
+
+- Node controller
+- Route controller
+- Service controller
+
+#### Node controller
+
+Node controller có vai trò khởi tạo một Node bằng cách thu thập thông tin về những Node đang chạy trong cụm từ các cloud provider.
+
+Node controller sẽ thực hiện những chức năng sau:
+
+1. Khởi tạo một Node với các nhãn region/zone.
+2. Khởi tạo một Node với những thông tin được cung cấp từ cloud, ví dụ như loại máy và kích cỡ.
+3. Thu thập địa chỉ mạng của Node và hostname.
+4. Trong trường hợp một Node không có tín hiệu phản hồi, Node controller sẽ kiểm tra xem Node này có thực sự xóa khỏi hệ thống cloud hay chưa. Nếu Node đó không còn tồn tại trên cloud, bộ controller sẽ xóa Node đó khỏi cụm Kubernetes.
+
+#### Route controller
+
+Route controller đóng vai trò cấu hình định tuyến trong nằm trong hệ thống cloud để các container trên các Node khác nhau trong cụm Kubernetes có thể giao tiếp với nhau. Bộ Route controller hiện chỉ đáp ứng được cho các cụm Google Compute Engine.
+
+#### Service controller
+
+Service controller lắng nghe các sự kiện tạo mới, cập nhật và xoá bỏ một service. Dựa trên trạng thái hiện tại của các vụ trên Kubernetes, nó cấu hình các bộ cân bằng tải trên cloud (như ELB của AWS, Google Load Balancer, hay Oracle Cloud Infrastructure LB) nhằm phản ánh trạng thái của các Service trên Kubernetes. Ngoài ra, nó đảm bảo những service backends cho các bộ cần bằng tải trên cloud được cập nhật
+
+### 2. Kubelet
+
+Node controller bao gồm một số tính nằng phụ thuộc vào tầng cloud của Kubelet. Trước khi có bộ CCM, Kubelet đảm nhận vai trò khởi tạo một Node với thông tin chi tiết từ cloud như địa chỉ IP, region hay instance type. Với bộ CCM, vai trò này được bộ CCM đảm nhận thay cho Kubelet.
+
+Với mô hình mới này, Kubelet sẽ khởi tạo một Node nhưng không đi kèm với những thông tin từ cloud. Tuy nhiên, nó sẽ thêm vào một dấu chờ (taint) đánh dấu Node sẽ không được lập lịch cho tới khi bộ CCM khởi tạo xong Node này với những thông tin cụ thể cung cấp từ Cloud, sau đó nó sẽ xóa những dấu chờ này.
```

Comment:
Please keep "dấu chờ (taint)" like the original: "a Taint". For more info: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/

Reply:
You could also add a glossary tooltip for taint, to help non-English readers understand what this is.

Reply:
I will. Thank for suggesting!

Reply:
@locmai : Let's refer this: 
{{< glossary_tooltip term_id="taint" text="Taint" >}}

Reply:
@sftim : Can you tell me where I can find list term_id in k8s?

Reply:
Let me try to use the tooltip markdown, not sure if I get it right XD

Reply:
For a list of glossary `term_id`s, see the metadata in files inside https://github.com/kubernetes/website/tree/master/content/en/docs/reference/glossary

For example:
https://raw.githubusercontent.com/kubernetes/website/a5a39ae8b1c8db6b0b997a90698a55ea11516fcc/content/en/docs/reference/glossary/workload.md

:information_source: _note_ that the `term_id` for that entry is “workloads”, not “Workload” (the English-localization title) or “workload” (from `workload.md`)

Reply:
It seems like we have to define the glossary page in Vietnamese for using the terms, or else it would fail the build. Is it correct?

Reply:
Done guys. But the Taint glossary includes some other terms. I will go back to update the terms later!

Reply:
I also initialized the reference/glossary directory too.

Reply:
Great, @locmai